public/app/features/dashboard/panel_editor/DataSourceOption.tsx

```diff
@@ -1,16 +1,17 @@
-import React, { FC } from 'react';
+import React, { FC, ChangeEvent } from 'react';
 import { FormLabel } from '@grafana/ui';
 
 interface Props {
   label: string;
   placeholder?: string;
-  name?: string;
-  value?: string;
-  onChange?: (evt: any) => void;
+  name: string;
+  value: string;
+  onBlur: (event: ChangeEvent<HTMLInputElement>) => void;
+  onChange: (event: ChangeEvent<HTMLInputElement>) => void;
   tooltipInfo?: any;
 }
 
-export const DataSourceOptions: FC<Props> = ({ label, placeholder, name, value, onChange, tooltipInfo }) => {
+export const DataSourceOption: FC<Props> = ({ label, placeholder, name, value, onBlur, onChange, tooltipInfo }) => {
   return (
     <div className="gf-form gf-form--flex-end">
       <FormLabel tooltip={tooltipInfo}>{label}</FormLabel>
```

Comment:
Should we use Grafanas Input instead?

Reply:
Good idea! We can probably use the same as the Gauge? https://github.com/grafana/grafana/blob/590450291af2c4c15593923115f8159a786cef7c/public/app/plugins/panel/gauge/GaugeOptionsBox.tsx#L31

Reply:
Created issue for this #15785